americangut/diversity_analysis.py

```diff
@@ -0,0 +1,1729 @@
+    """Wraps matplotlib's heatmap and formats the colorbar
+
+    Parameters
+    ----------
+    data : array
+        The data to be plotted in the heat map
+   ax : matplotlib axis, optional
+        The axis where data should be plotted. If none, a new axis instance
+        will be created.
+    xticklabels : list, optional
+        The labels to be used for the x-axis, to describe the groups of
+        data.
+    yticklabels : list, optional
+        The labels for each group on the y-axis.
+    cmap : str, optional
+        The colormap name to use for plotting.
+    clims : list, optional
+        The limits on the colormap
+    xfont_angle : float, optional
+        The angle in degrees for the x tick label text.
+    xfont_align : {'left', 'right', 'center'}, optional
+        The horizonal alignment of the x tick label text. For rotated text,
+        an alignment or 'right' is recommended.
+    yfont_angle : float, optional
+        The angle in degrees for the x tick label text.
+    yfont_align : {'left', 'right', 'center'}, optional
+        The horizonal alignment of the x tick label text. For rotated text,
+        an alignment or 'right' is recommended.
+    xfont_size : int, optional
+        Default is 12. The size of the x tick labels
+    yfont_size : int, optional
+        Default is 12. The size of the y tick labels
+    cbar_size : int, optional
+        Default is 11. The size of the labels on the colorbar.
+
+    Returns
+    -------
+    ax: matplotlib axis
+        The axis with the data plotted.
+    cbar : matplotlib colorbar
+        The colorbar instance on the plot
+
+    """
+    # Checks the shape of the data is sane
+    mat_shape = data.shape
+    if ('xticklabels' in kwargs and kwargs['xticklabels'] is not None and not
+        len(kwargs['xticklabels']) == mat_shape[1]):
+        raise ValueError('There must be a label for each column in '
+                         'data.')
+
+    if ('yticklabels' in kwargs and kwargs['yticklabels'] is not None and not
+        len(kwargs['yticklabels']) == mat_shape[0]):
+        raise ValueError('There must be a label for each row in data.')
+
+    # Gets the axis
+    if ax is None:
+        ax = plt.axes()
+    # Gets the associated figure
+    fig = ax.get_figure()
+    # Plots the data
+    im = ax.pcolor(data, cmap=cmap)
+    if clims is not None:
+        im.set_clim(clims)
+    cbar = fig.colorbar(im, ax=ax)
+
+    # Gets the axis limits
+    if 'xlim' not in kwargs:
+        kwargs['xlim'] = [0, mat_shape[1]]
+    if 'ylim' not in kwargs:
+        kwargs['ylim'] = [mat_shape[0], 0]
+    if 'xticks' not in kwargs:
+        kwargs['xticks'] = np.arange(0, mat_shape[1]) + 0.5
+    if 'yticks' not in kwargs:
+        kwargs['yticks'] = np.arange(0, mat_shape[0]) + 0.5
+
+    # Formats the axis
+    _format_axis(ax, **kwargs)
+
+    # Formats the colorbar ylabels
+    new_labels = ['%s' % t.get_text() for t in cbar.ax.get_yticklabels()]
```

Comment:
This converts the labels from a text object to a string that can be operated on. Removing the `$` lets the font for the color bar match the rest of the fonts in the figure if using LaTeX for font rendering.